lua/nvim-treesitter/textobjects/select.lua

```diff
@@ -152,9 +152,11 @@ function M.attach(bufnr, lang)
       query = nil
     end
     if query then
-      vim.keymap.set({ "o", "x" }, mapping, function()
-        require("nvim-treesitter.textobjects.select").select_textobject(query)
-      end, { buffer = buf, silent = true, remap = false, desc = desc })
+      for _, mode in ipairs { "o", "x" } do
+        vim.keymap.set(mode, mapping, function()
+          require("nvim-treesitter.textobjects.select").select_textobject(query, mode)
+        end, { buffer = buf, silent = true, remap = false, desc = desc })
+      end
     end
```

Comment:
Theseines are equivalent: vim.keymap.set can take the modes as a table.

The mapping exists, but it doesn't do what it used to. If I'm not mistaken, using a callback with vim.keymap.set in visual mode requires taking some steps in the callback, meaning the same callback can't really be used (unless it inspects the parameters to check the mode).

Reply:
Hmm maybe there are more issues with this but at least the `mode` is now passed to `select_textobject` as it was before the change to `vim.keymap.set`.

Reply:
So we should just go back to what it was before?

Reply:
> Hmm maybe there are more issues with this but at least the `mode` is now passed to `select_textobject` as it was before the change to `vim.keymap.set`.

Oh interesting, I wonder if there's a bug in neovim upstream when passing multiple modes to vim.keymap.set? 🤔